tests/test_repository.py

```diff
@@ -140,6 +143,19 @@ def test_clone_from_space(self):
                 use_auth_token=self._token,
             )
 
+    def test_clone_from_deprecation_warning(self):
+        """SKIP CREATION"""
+        with pytest.raises(
+            FutureWarning,
+            match="Creating a repository through clone_from is deprecated*",
+        ):
+            Repository(
+                WORKING_REPO_DIR,
+                clone_from=f"{USER}/{uuid.uuid4()}",
+                use_auth_token=self._token,
+                repo_type="dataset",
```

Comment:
Nit: feel free to remove `repo_type`. The simpler the test the easier to understand, so anything not strictly needed I would remove

Reply:
I did it because there was two of those tests, one was for model repository and one was for dataset repository.

Reply:
I think that given the behaviour you're changing is ok to just cover one case